ui/src/app/views/workspace-list/component.tsx

```diff
@@ -0,0 +1,350 @@
+import {Component} from '@angular/core';
+import {ErrorHandlingService} from 'app/services/error-handling.service';
+
+import {navigate} from 'app/utils/navigation';
+import {WorkspacePermissions} from 'app/utils/workspace-permissions';
+
+import {AlertDanger, AlertWarning} from 'app/components/alert';
+import {Button, CardButton, Clickable, Link, MenuItem} from 'app/components/buttons';
+import {Card} from 'app/components/card';
+import {FadeBox} from 'app/components/containers';
+import {ListPageHeader} from 'app/components/headers';
+import {ClrIcon} from 'app/components/icons';
+import {Modal, ModalBody, ModalFooter, ModalTitle} from 'app/components/modals';
+import {PopupTrigger, TooltipTrigger} from 'app/components/popups';
+import {Spinner} from 'app/components/spinners';
+import {workspacesApi} from 'app/services/swagger-fetch-clients';
+import {displayDate, reactStyles, ReactWrapperBase, withUserProfile} from 'app/utils/index';
+import {BugReportModal} from 'app/views/bug-report/component';
+import {ConfirmDeleteModal} from 'app/views/confirm-delete-modal/component';
+import {WorkspaceShare} from 'app/views/workspace-share/component';
+import {
+  BillingProjectStatus,
+  ErrorResponse,
+  Profile,
+} from 'generated/fetch';
+import * as React from 'react';
+
+const styles = reactStyles({
+  fadeBox: {
+    margin: '1rem auto 0 auto', width: '97.5%', padding: '0 1rem'
+  },
+  cardArea: {
+    display: 'flex', justifyContent: 'flex-start', flexWrap: 'wrap'
+  },
+  addCard: {
+    display: 'flex', fontSize: '20px', lineHeight: '28px', marginTop: '0',
+    fontWeight: 600, color: '#216FB4'
+  },
+  workspaceName: {
+    color: '#216FB4', marginBottom: '0.5rem', fontWeight: 600,
+    fontSize: 18, wordBreak: 'break-all', cursor: 'pointer',
+  },
+  workspaceDescription: {
+    textOverflow: 'ellipsis', overflow: 'hidden', height: '2rem', display: '-webkit-box',
+    WebkitLineClamp: 2, WebkitBoxOrient: 'vertical'
+  },
+  workspaceCard: {
+    display: 'flex', flexDirection: 'column', justifyContent: 'space-between', height: '100%'
+  },
+  workspaceCardFooter: {
+    display: 'flex', flexDirection: 'row', justifyContent: 'space-between', alignItems: 'center'
+  },
+  permissionBox: {
+    color: '#FFFFFF', height: '1rem', width: '3rem', fontSize: 10, textAlign: 'center',
+    borderRadius: '0.2rem', padding: 0
+  }
+});
+
+
+const WorkspaceCardMenu: React.FunctionComponent<{
+  disabled: boolean, wp: WorkspacePermissions, onShare: Function, onDelete: Function
```

Comment:
There's something funky going on with the formatting of these couple lines, made it difficult to parse.